libr/anal/p/anal_x86_cs.c

```diff
@@ -930,8 +930,13 @@ static void anop_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len,
 	case X86_INS_PUSH:
 		{
 			dst = getarg (&gop, 0, 0, NULL, DST_AR);
-			esilprintf (op, "%s,%d,%s,-=,%s,=[%d]",
-				dst?dst:"eax", rs, sp, sp, rs);
+			if (strcmp (dst, sp)) {
```

Comment:
better use !strcmp, default cases are expected to be the ones that match to make it more readable